plugins/modules/supervisorctl.py

```diff
@@ -214,7 +223,8 @@ def take_action_on_processes(processes, status_filter, action, expected_result):
             if '%s: %s' % (process_name, expected_result) not in out:
                 module.fail_json(msg=out)
 
-        module.exit_json(changed=True, name=name, state=state, affected=to_take_action_on)
+        if exit_module:
+            module.exit_json(changed=True, name=name, state=state, affected=to_take_action_on)
```

Comment:
There are also `exit_json` calls in lines 218 and 220.

Reply:
Fixed. This is my first time contributing, thank you for your patience :).